tests/test_vpc.py

```diff
@@ -559,6 +562,133 @@ def test_interface_subnet(self):
+
+        return instance_data
+
+    def allocate_network_address(self, ec2, domain_type):
+        network_addr = ec2.allocate_address(Domain=domain_type)
+        self.addCleanup(self.release_if_still_present, ec2, network_addr)
+        return network_addr
+
+    def core_release_op(self, factory, ec2, network_addr, force=False):
+        alloc_id = network_addr['AllocationId']
+
+        p = self.load_policy({
+            'name': 'release-network-addr',
+            'resource': 'network-addr',
+            'filters': [
+                {'AllocationId': alloc_id},
+            ],
+            'actions': [{'type': 'release', 'force': force}], },
+            session_factory=factory)
+
+        resources = p.run()
+
+        self.assertEqual(len(resources), 1)
+        with self.assertRaises(BotoClientError) as e_cm:
+            ec2.describe_addresses(AllocationIds=[ alloc_id ])
+        e = e_cm.exception
+        self.assertEqual(e.response['Error']['Code'], 'InvalidAllocationID.NotFound')
+        self.assertIn(alloc_id, e.response['Error']['Message'])
+
+    @functional
+    def test_release_detached_vpc(self):
+        factory = self.replay_flight_data('test_release_detached_vpc')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        network_addr = self.allocate_network_address(ec2, 'vpc')
+        self.core_release_op(factory, ec2, network_addr)
+
+    @functional
+    def test_release_detached_classic(self):
+        factory = self.replay_flight_data('test_release_detached_classic')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        network_addr = self.allocate_network_address(ec2, 'classic')
+        self.core_release_op(factory, ec2, network_addr)
+
+    # This is not a functional test because spinning up the ec2 takes too long
+    def test_release_attached_ec2_vpc(self):
+        factory = self.replay_flight_data('test_release_attached_ec2_vpc')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        ec2_instance = self.create_ec2_instance(ec2)
+        network_addr = self.allocate_network_address(ec2, 'vpc')
+
+        assoc_spec = AddressRelease.gen_assoc_spec(network_addr)
+        assoc_spec['InstanceId'] = ec2_instance['InstanceId']
+        assoc_resp = ec2.associate_address(**assoc_spec)
+        network_addr_attached = ec2.describe_addresses(AllocationIds=[ network_addr['AllocationId'] ])
+        self.assertEqual(len(network_addr_attached['Addresses']), 1)
+        self.assertEqual(network_addr_attached['Addresses'][0]['AssociationId'], assoc_resp['AssociationId'])
+
+        self.core_release_op(factory, ec2, network_addr, True)
+
+    def test_release_attached_nif_vpc(self):
+        factory = self.replay_flight_data('test_release_attached_nif_vpc')
+
+        session = factory()
+        ec2 = session.client('ec2')
+
+        subnets = ec2.describe_subnets(Filters=[{"Name": "defaultForAz", "Values": [ "true" ] }])
```

Comment:
since we're not in functional mode, its fine to just put in the subnet id into the call on create network interface to avoid the extra describe.